File: 1.cpp
#include <iostream>
#include <vector>
using namespace std;

void selection_sort(vector<int> &arr) {
    int n = arr.size();
    for (int i = 0; i < n; i++) {
        int min_index = i;
        for (int j = i + 1; j < n; j++) {
            if (arr[j] < arr[min_index]) {
                min_index = j;
            }
        }
        swap(arr[i], arr[min_index]);
    }
}

void bubble_sort(vector<int> &arr) {
    int swapped;
    int size = sizeof(arr)/ sizeof(arr[0]);
    int n = size - 1;

    do {
        swapped = 0;
        for (int j = 0; j < n; j++) {
            if (arr[j] > arr[j + 1]) {
                int temp = arr[j];
                arr[j] = arr[j + 1];
                arr[j + 1] = temp;
                swapped = 1;
            }
        }
        n--;
    } while (swapped);
}


void print_array(const vector<int> &arr) {
    for (int num : arr) {
        cout << num << " ";
    }
    cout << endl;
}

int main() {
    vector<int> array = {10, 7, 8, 9, 1, 5};

    selection_sort(array);
    print_array(array);

    bubble_sort(array);
    print_array(array);

    return 0;
}

File: 1.js
function selectionSort(arr) {
    const n = arr.length;
    for (let i = 0; i < n; i++) {
        let min_index = i;
        for (let j = i + 1; j < n; j++) {
            if (arr[j] < arr[min_index]) {
                min_index = j;
            }
        }
        [arr[i], arr[min_index]] = [arr[min_index], arr[i]];
    }
    return arr;
}

function bubbleSort(arr) {
    let swapped;
    let n = arr.length - 1;

    do {
        swapped = false;
        for (let j = 0; j < n; j++) {
            if (arr[j] > arr[j + 1]) {
                [arr[j], arr[j + 1]] = [arr[j + 1], arr[j]];
                swapped = true;
            }
        }
        n--;
    } while (swapped);

    return arr;
}

const array = [10, 7, 8, 9, 1, 5];

console.log(selectionSort(array));

console.log(bubbleSort(array));